File: packages/android/app/src/main/java/studio/eugenezakharov/opencode/api/models/AgentInfo.kt
package studio.eugenezakharov.opencode.api.models

import kotlinx.serialization.Serializable

/**
 * A selectable AI agent, from `GET /agent`. `mode` is subagent / primary / all;
 * only primary/all (non-hidden) agents are user-selectable. Mirrors iOS `AgentInfo`.
 */
@Serializable
data class AgentInfo(
    val name: String,
    val description: String? = null,
    val mode: String = "primary",
    val hidden: Boolean = false,
) {
    val selectable: Boolean get() = mode != "subagent" && !hidden
}

File: packages/ios/OpenCode/API/Models/AgentInfo.swift
import Foundation

/// A selectable AI agent, from `GET /agent`. `mode` is subagent / primary / all;
/// only primary/all (non-hidden) agents are user-selectable in the composer.
struct AgentInfo: Decodable, Identifiable {
    let name: String
    let description: String?
    let mode: String
    let hidden: Bool?

    var id: String { name }
    var selectable: Bool { mode != "subagent" && !(hidden ?? false) }
}
